packages/lesswrong/server/authenticationMiddlewares.ts

```diff
@@ -124,20 +172,28 @@ export const addAuthMiddlewares = (addConnectHandler) => {
   addConnectHandler('/logout', (req, res, next) => {
     passport.authenticate('custom', (err, user, info) => {
       if (err) return next(err)
-      if (!user) return next()
       req.logOut()
 
-      // The accepted way to delete a cookie is to set an expiration date in the past.
-      if (getCookieFromReq(req, "meteor_login_token")) {
-        res.setHeader("Set-Cookie", `meteor_login_token= ; expires=${new Date(0).toUTCString()};`)
+      // Remove session cookies
+      const cookieUpdates = ['meteor_login_token', 'loginToken', 'connect.sid']
+        .filter(cookieName => getCookieFromReq(req, cookieName))
+        // The accepted way to delete a cookie is to set an expiration date in the past.
+        .map(cookieName => `${cookieName}= ; expires=${new Date(0).toUTCString()}`)
+      if (cookieUpdates.length) {
+        res.setHeader('Set-Cookie', cookieUpdates)
       }
-      if (getCookieFromReq(req, "loginToken")) {
-        res.setHeader("Set-Cookie", `loginToken= ; expires=${new Date(0).toUTCString()};`)
-      }
-      
       
       res.statusCode=302;
-      res.setHeader('Location','/');
+      // Need to log the user out of their Auth0 account. Otherwise when they
+      // next try to login they won't be given a choice, just auto-resumed to
+      // the same Auth0 account.
+      if (auth0DomainSetting.get() && auth0ClientIdSetting.get()) {
```

Comment:
You're testing if these vars are set, but that condition doesn't necessarily map 1-1 with a site that's exclusively using Auth0, does it? Like, it does in reality at the moment, but I guess you could imagine that LW adds 'Login with your EA.org' account as an option, in which case not every user should go through this flow, right? Maybe being way to defensive here, and there probably aren't big drawbacks to doing this, but I guess if there's an easy test to see if the user is an Auth0 user then that might be a more semantically stable test.

Reply:
If less wrong adds that, logging out of lesswrong probably doesn't mean that they should log out of their EA.org account, even if the user was an Auth0 user. I've added a check for the forumType.